source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -316,50 +316,57 @@ def onPanelActivated(self):
 		self.lastControl = self.optionsText
 
 	def _updateEnabledState(self):
-		settings = self._getSettingsStorage()
+		settingsStorage = self._getSettingsStorage()
 		settingsToTriggerActivation = [
-			settings.highlightBrowseMode,
-			settings.highlightFocus,
-			settings.highlightNavigator,
+			settingsStorage.highlightBrowseMode,
+			settingsStorage.highlightFocus,
+			settingsStorage.highlightNavigator,
 		]
-		if any(settingsToTriggerActivation):
-			if all(settingsToTriggerActivation):
-				self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
-			else:
-				self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
-			self._ensureEnableState(True)
+		isAnyEnabled = any(settingsToTriggerActivation)
+		if all(settingsToTriggerActivation):
+			self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
+		elif isAnyEnabled:
+			self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
 		else:
 			self._enabledCheckbox.Set3StateValue(wx.CHK_UNCHECKED)
-			self._ensureEnableState(False)

```

Comment:
This logic is intended to be the same, but removes a level of nesting, and avoids having to use explicit True / False literal for `_ensureEnabledState`